tasks/perform_orca_reconcile/perform_orca_reconcile.py

```diff
@@ -438,7 +452,7 @@ def handler(
         db_connect_info_secret_arn = os.environ["DB_CONNECT_INFO_SECRET_ARN"]
     except KeyError as key_error:
         LOGGER.error("DB_CONNECT_INFO_SECRET_ARN environment value not found.")
-        raise
+        raise key_error
```

Comment:
I recommend reverting this line, and removing the unused variable `key_error`. I don't know if it's the same in Python, but in some languages manually re-raising an error messes with the stack trace.